src/Compilers/CSharp/Test/Emit/Emit/DynamicAnalysis/DynamicInstrumentationTests.cs

```diff
@@ -155,45 +158,53 @@ static void Fred()
 ";
 
             string expectedBarneyIL = @"{
-  // Code size       83 (0x53)
+  // Code size       82 (0x52)
   .maxstack  4
-  IL_0000:  ldsfld     ""bool[] Program.<Barney>3ipayload__Field""
-  IL_0005:  brtrue.s   IL_001c
-  IL_0007:  ldsfld     ""System.Guid <PrivateImplementationDetails>.MVID""
-  IL_000c:  ldtoken    ""int Program.Barney(bool)""
-  IL_0011:  ldsflda    ""bool[] Program.<Barney>3ipayload__Field""
-  IL_0016:  ldc.i4.5
-  IL_0017:  call       ""void Microsoft.CodeAnalysis.Runtime.Instrumentation.CreatePayload(System.Guid, int, ref bool[], int)""
-  IL_001c:  ldsfld     ""bool[] Program.<Barney>3ipayload__Field""
-  IL_0021:  ldc.i4.1
-  IL_0022:  ldc.i4.1
-  IL_0023:  stelem.i1
-  IL_0024:  ldarg.0
-  IL_0025:  brfalse.s  IL_0032
-  IL_0027:  ldsfld     ""bool[] Program.<Barney>3ipayload__Field""
-  IL_002c:  ldc.i4.0
-  IL_002d:  ldc.i4.1
-  IL_002e:  stelem.i1
-  IL_002f:  ldc.i4.s   10
-  IL_0031:  ret
-  IL_0032:  ldsfld     ""bool[] Program.<Barney>3ipayload__Field""
-  IL_0037:  ldc.i4.3
+  .locals init (bool[] V_0)
```

Comment:
Thanks for reading the IL! I've wrestled with this point without coming to a definite conclusion. It's an IL size vs. execution optimization question. We have a design goal of minimizing the execution time overhead of instrumentation. By inlining the null check, all invocations of a method except the first have a very low overhead: load an array element, test against null, and branch. Adding in a method call to the prologue for every invocation probably noticeably increases the execution cost of small methods.